primitives/state-machine/src/lib.rs

```diff
@@ -347,11 +351,11 @@ impl<'a, B, H, N, Exec> StateMachine<'a, B, H, N, Exec> where
 				CallResult<R, Exec::Error>,
 			) -> CallResult<R, Exec::Error>
 	{
-		let pending_changes = self.overlay.clone_pending();
+		self.overlay.start_transaction();
```

Comment:
> Isn't this covered by Ext::new call in execute_aux?

`enter_runtime` does not start a new transaction. It merely protects the existing ones from being dropped.

> Ok reading a bit more, we need to start a transaction as runtime entering action do not start one.

You know you can edit your comments? You do not need to put your whole thought process into the comment. I have a hard time reading your comments which results in frustration on my site.

> it seems like all execute_aux are under a transaction 

Only native executions because there we intent to roll them back in order to re-run them in wasm when necessary.

Reply:
>  You do not need to put your whole thought process into the comment

will try, I usually have a problem with delete, at least the thought process in question is here to ask the question 'why not binding transaction start with runtime enter'? But it is redundant with the following comments to.
Actually got a repy from previous comment (no need to reply).